python-frontend/src/main/java/org/sonar/plugins/python/api/tree/LineMagic.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.api.tree;
+
+public interface LineMagic extends Expression {
```

Comment:
I would expect this to be an `ExpressionStatement`, is this done intentionally?

Reply:
Line magic could be part of an expression as well as a statement like:
```python
# line magic statement
%time a = foo()
# lime magic expression
a = %time foo()
```

That's why I've implemented it as LineMagic which is Expression and LineMagicStatement which is a statement containing an expression

Reply:
As we discussed, we don't actually have nodes extending `ExpressionStatement` (except for `ExpressionStatementImpl`). The usual way we've dealt with expression statements (like `CallExpression`) is by having them defined as `Expression` and nesting them into an `ExpressionStatement`.
I expect this is what we should do here as well (and thus, ditch the concept of `LineMagicStatement`).